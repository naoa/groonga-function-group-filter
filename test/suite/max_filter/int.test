plugin_register functions/group_filter
table_create docs TABLE_HASH_KEY ShortText
column_create docs num COLUMN_SCALAR Int64
load --table docs
[
{"_key": "JP2001213456", "num": "2020"},
{"_key": "JP2001213457", "num": "2012"},
{"_key": "JP2001213458", "num": "2016"}
]

table_create num TABLE_HASH_KEY Int64
column_create num num COLUMN_INDEX docs num

select docs \
  --filter '_id > 0 && max_filter("num", 5)' \
  --output_columns '_id,_score,_key,num' --command_version 2
