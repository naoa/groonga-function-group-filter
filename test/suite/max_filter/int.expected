plugin_register functions/group_filter
[[0,0.0,0.0],true]
table_create docs TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create docs num COLUMN_SCALAR Int64
[[0,0.0,0.0],true]
load --table docs
[
{"_key": "JP2001213456", "num": "2020"},
{"_key": "JP2001213457", "num": "2012"},
{"_key": "JP2001213458", "num": "2016"}
]
[[0,0.0,0.0],3]
table_create num TABLE_HASH_KEY Int64
[[0,0.0,0.0],true]
column_create num num COLUMN_INDEX docs num
[[0,0.0,0.0],true]
select docs   --filter '_id > 0 && max_filter("num", 5)'   --output_columns '_id,_score,_key,num' --command_version 2
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        2
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_score",
          "Float"
        ],
        [
          "_key",
          "ShortText"
        ],
        [
          "num",
          "Int64"
        ]
      ],
      [
        1,
        1.0,
        "JP2001213456",
        2020
      ],
      [
        3,
        1.0,
        "JP2001213458",
        2016
      ]
    ]
  ]
]
